src/test/groovy/com/ss/mqtt/broker/test/integration/ConnectSubscribePublishTest.groovy

```diff
@@ -12,17 +12,30 @@ class ConnectSubscribePublishTest extends MqttBrokerTest {
         given:
             Mqtt5Publish receivedMessage = null
         when:
-           
-            mqttSubscriber.connect().join()
-            mqttPublisher.connect().join()
             
+            mqttSubscriber.connectWith()
```

Comment:
I think you need to test authentication in ConnectTest, it's not part of publish/subscribe